### PR TITLE
feat(service-portal): mobile and user menu UI changes - 2

### PR DESCRIPTION
**Pull request by disaerna**
# Mobile menu and User menu changes

_New pullrequest - This PR #6186  was keeping commit history from other branch_

## What

Remove modal on mobile and fullscreen screen instead

## Why

Design modified to be more consistent with main web

Mobile menu displays language button and user menu button with circle X to close
User menu displays over Mobile menu with the same language button and circle X to close

## Screenshots / Gifs

Screenshot 2022-01-10 at 10 45 55

Screenshot 2022-01-10 at 10 45 47

Screenshot 2022-01-10 at 10 45 42

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thordurhhh — APPROVED** [review]
LGTM 💪

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx [hunk 1]
```diff
@@ -14,17 +26,26 @@ export const UserMenu = ({ fullscreen = false }: { fullscreen?: boolean }) => {
...
   }
+  useEffect(() => {
+    setUserMenuOpen && setUserMenuOpen(dropdownState === 'open' ? true : false)
+  }, [dropdownState])
+
+  useEffect(() => {
+    if ((dropdownState === 'open') !== userMenuOpen) {
+      setDropdownState('closed')
+    }
+  }, [userMenuOpen])
```
Comment (saevarma) [inline libs/shared/components/src/auth/UserMenu/UserMenu.tsx:37]: I find this a bit confusing. if `dropdownState = closed` and `userMenuOpen = true` then we will get `false !== true` and set the dropdown state to `closed` but it is already closed. Shouldn't it be changed to open?
And also why have two variables keeping the open/closed state instead of just exposing the dropdownState?

File: apps/service-portal/src/components/Sidebar/Sidebar.css.ts [hunk 2]
```diff
@@ -43,7 +42,7 @@ export const subnavCollapsed = style({
 export const navIcon = style({
   position: 'absolute',
   right: -theme.spacing[4],
-  top: theme.spacing[10],
+  top: 43,
```
Comment (saevarma) [inline apps/service-portal/src/components/Sidebar/Sidebar.css.ts:45]: Shouldn't this be based on the theme? `43` seems a bit random number, it is not fitting into the multiplier of 8

**Review by saevarma — CHANGES_REQUESTED** [review]
Few minor comments.

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx (hunk 1 above)
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx:37]: Yes it is confusing! 🤯  took me a while to understand it myself 😆 

"userMenuOpen" and "setUserMenuOpen" is to control the dropdown state as well. But I have to access it through the Header component. 
The new design has this language button and X button that is displayed when a user has the "UserMenu" open, but only in service portal. 

This check is to make sure that the X button (that is displayed in the Header component) is closing the UserMenu when the dropdownState is open.
Basically if I press X then the state changes and I check if the dropdown is open, if so I close it. 
I changed the code to this - hopefully more readable? : 

`useEffect(() => {
    if (dropdownState === 'open' && !userMenuOpen) {
      setDropdownState('closed')
    }
  }, [userMenuOpen])
`

This is all getting quite bloated & confusing and it is time to refactor it all in the next refactor week! I also wonder if it is better to use separate component in the Service Portal 🤔

File: libs/shared/components/src/auth/UserMenu/UserMenu.spec.tsx [hunk 3]
```diff
@@ -193,6 +193,56 @@ describe('UserMenu', () => {
...
+    const languageSelector = screen.getByTestId('language-switcher-button')
+    expect(languageSelector).not.toBeNull()
+    expect(screen.getByText(/Current/)).toHaveTextContent('Current: is')
+    // Act
+    fireEvent.click(screen.getByText('EN'))
+
+    // Assert
+    expect(screen.getByText(/Current/)).toHaveTextContent('Current: en')
+  })
+  it('can switch languages using selectbox', async () => {
```
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.spec.tsx:217]: Not really - one is rendering select box language switcher and the other one just a button.

File: apps/service-portal/src/components/Header/Header.tsx [hunk 4]
```diff
@@ -1,31 +1,55 @@
...
   const handleMobileMenuTriggerClick = () =>
     dispatch({
       type: ActionType.SetMobileMenuState,
       payload: mobileMenuState === 'open' ? 'closed' : 'open',
     })
 
+  const closeButton = (userMenu: boolean) => {
+    return (
+      <Box display="flex" alignItems="center">
+        <FocusableBox
```
[the remark this answers is not in the document]
Comment (disaerna) [reply apps/service-portal/src/components/Header/Header.tsx:36]: fixed

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx [hunk 5]
```diff
@@ -14,17 +26,26 @@ export const UserMenu = ({ fullscreen = false }: { fullscreen?: boolean }) => {
   const handleClick = () => {
     setDropdownState(dropdownState === 'open' ? 'closed' : 'open')
   }
+  useEffect(() => {
+    setUserMenuOpen && setUserMenuOpen(dropdownState === 'open' ? true : false)
```
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx:30]: fixed

File: apps/service-portal/src/components/Sidebar/Sidebar.css.ts (hunk 2 above)
Comment (disaerna) [reply apps/service-portal/src/components/Sidebar/Sidebar.css.ts:45]: Yes it should, but than it isn't in the center 😞

**Review by saevarma — CHANGES_REQUESTED** [review]

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx (hunk 1 above)
Comment (saevarma) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx]: I see and yes this is more readable.

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx [hunk 6]
```diff
@@ -14,17 +26,27 @@ export const UserMenu = ({ fullscreen = false }: { fullscreen?: boolean }) => {
   const handleClick = () => {
     setDropdownState(dropdownState === 'open' ? 'closed' : 'open')
   }
+  useEffect(() => {
+    setUserMenuOpen && setUserMenuOpen(dropdownState === 'open')
+  }, [dropdownState])
+
+  useEffect(() => {
+    // false !==
```
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx:34]: ✅

**Review by saevarma — APPROVED** [review]
